batch.go

```diff
@@ -107,14 +107,12 @@ func (br *batchResults) Query() (Rows, error) {
 		query = "batch query"
 	}
 
-	rows := br.conn.getRows(br.ctx, query, arguments)
-
 	if br.err != nil {
-		rows.err = br.err
-		rows.closed = true
-		return rows, br.err
+		return &connRows{err: br.err, closed: true}, br.err
 	}
 
+	rows := br.conn.getRows(br.ctx, query, arguments)
+
```

Comment:
This is the real change. It simply moves the error handling above the call to `getRows` and returns a bare-minimum `*connRows` instance.